### ISSUE TITLE
Improve package management design documentation

### ISSUE THREAD

**Issue by kolbe** [pingcap/tiup#1247]
## Feature Request

**Is your feature request related to a problem? Please describe:**

It's currently very difficult to understand the design of TiUP with respect to the handling of nightly builds, component discovery, etc.

TiUP uses a nightly build tagging system that apparently doesn't match the tagging system used in downloadable packages, Docker images, or elsewhere. The TiUP package discovery and management system should be fully documented.

**Describe the feature you'd like:**

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

**Comment by lucklove** [comment]
@King-Dylan

**Comment by King-Dylan** [comment]
Adding this logic to tiup will cause compatibility issues ，and @zhouqiang-cl will to optimize nightly build tagging system in the next two to three months

**Comment by lucklove** [comment]
> Adding this logic to tiup will cause compatibility issues ，and @zhouqiang-cl will to optimize nightly build tagging system in the next two to three months

I think what kolbe says is to improve document, not change current behavior

**Comment by kolbe** [comment]
Correct, this is a request for improved documentation, not for any changes to the product.
